File: lib/compile_systems/cpp_system.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main(int argc, char const *argv[])
{
  ofstream ofs;
  ofs.open("test.cpp",ios::binary);
  string buf;
  while(getline(cin,buf)){
    if(buf=="<$><*><$><*><$><*><$><*><$><*><$><*><$>")break;
    ofs<<buf<<endl;
  }
  ofs.close();
  ofs.open("test.in",ios::binary);
  while(getline(cin,buf)){
    ofs<<buf<<endl;
  }
  ofs.close();
  system("/opt/wandbox/gcc-head/bin/g++ test.cpp -o test -O2");
  system("/usr/bin/time -f '%U' ./test < test.in");
  return 0;
}

File: lib/compile_systems/cpp_system.rb
require 'zlib'

File.open("./main.cpp", "w") do |file|
	loop{
		str = gets.chomp.to_s
		break if str == "<$><*><$>"
		file.puts str
	}
	file.close
end

File.open("./test.in", "w") do |file|
	loop{
		str = gets.chomp.to_s
		break if str == "<$><*><$>"
		file.puts Zlib::Inflate.inflate(str)
	}
	file.close
end
time = gets
system "/opt/wandbox/gcc-head/bin/g++ main.cpp -o main -O2"
system "timeout -s 9 " + time + " ./main < test.in" 